src/extremes_based/extremesdim.jl

```diff
@@ -206,3 +207,44 @@ function extremevaltheory_local_dim_persistence(
     Δ, θ = extremevaltheory_local_dim_persistence(logdist, p; kw...)
     return Δ, θ
 end
+
+
+
+"""
+BMextremedimensions(x:: StateSpaceSet) -> D1, θ
+    Computation of the local dimensions D1 and the extremal index θ for each observation in the 
+    trajectory x. This function uses the block maxima approach: divides the data in blocks of 
+    length ⌊√(N-1)⌋, where N is the number of data and ⌊⋅⌋ is the floor function, and takes the maximum
+    of those bloks.
+    Note that this implies that the initial 2⌊√(N-1)⌋{√(N-1)} + {√(N-1)}^2 - 1 data are not used. 
+    To reduce the number of unused data chose the number of data close to a perfect square + 1. 
+    The extremal index can be interpreted as the inverse of the persistance of the extremes around
+    that point.
+    
+"""
+function BMextremedimensions(x:: StateSpaceSet)
+
+    println("Computing dynamical quantities")
+
+    N = length(x[:,1])
+    quanti = 1 - 1/sqrt(N) # Heuristic, probably not optimal 
+    blocksize = Int64(floor(sqrt(N - 1))) 
+    newN = blocksize^2 + 1
+    firstindex = N - newN + 1
+    D1 = zeros(newN);
+    θ = zeros(newN);
+    for (j, k) in enumerate(range(firstindex,N))
+        # Compute the observables
+        logdista = -log.([euclidean(x[k,:],x[i,:]) for i in range(firstindex,N)])
+        # Compute the extremal index, use the external function extremal_Sueveges
+        θ[j] = extremal_index_sueveges(logdista, quanti)
+        # Remove the inf data
+        deleteat!(logdista, j)
```

Comment:
If there are two identical points in `x`, then there will be at least one more `Inf` value here. What does this do to the `erstimate_gev_scale` estimation?

Reply:
If there is two identical points, the trajectory should be periodic, no? In that case the function will not work, the estimated parameters will be NaN and the dimension as well.

Reply:
Yeah, isn't this incorrect and a problem? Periodic trajectories have dimension 1, not NaN.

Reply:
Well, if it is a discrete system the trajectory would be 0th dimensional, and the algorithm could be modified to return 0, which in some sense it is what it is doing (returns `1/NaN`, the problem is that if there is an `Inf` value it computes `Inf-Inf` at some point and gives `NaN` for the parameter sigma, this former `NaN` could be turn into an `Inf`). If you are trying to compute the dimension of a continuous limit cycle for example, it shouldn't cause any problem unless you happen to have a step size such that the sum of the discretizations of the integration step up to some finite `n` is an exact multiple of the lenght of the cycle, which sounds quite a rare occurrence to me. Then the algorithm will return `NaN`, but you would be actually sampling this continuous limit cycle only in a finite set of points forever.

Reply:
We need to be practical here. If you have experimentally sampled data, let's say you record chemical concentrations or whatever, the data are very often truncated. You only have 4-5 digits of decimal. It is not impossible to have the same data point recorded twice, irrespectively of whether the data points would identical "in theory" or not. The question is what do we do about it in the software implementation. You could, as the code does not, return `NaN` if there is a duplicate data point. There are other options: to skip all `Inf` values. Or, to throw an error message that says "algorithm detected duplicate points. In this case σ parameter is computed as NaN. To counter this, you could add white noise of minimum strength to the data to ensure that no duplicate points exist."

There are three options here, and I am not sure that option 1 (returning `NaN`) is the most useful option.